MobileWallet/TariLib/Wrappers/Utils/TestnetKeyServer.swift

```diff
@@ -0,0 +1,178 @@
+//  TestnetKeyServer.swift
+
+/*
+	Package MobileWallet
+	Created by Jason van den Berg on 2020/02/07
+	Using Swift 5.0
+	Running on macOS 10.15
+
+	Copyright 2019 The Tari Project
+
+	Redistribution and use in source and binary forms, with or
+	without modification, are permitted provided that the
+	following conditions are met:
+
+	1. Redistributions of source code must retain the above copyright notice,
+	this list of conditions and the following disclaimer.
+
+	2. Redistributions in binary form must reproduce the above
+	copyright notice, this list of conditions and the following disclaimer in the
+	documentation and/or other materials provided with the distribution.
+
+	3. Neither the name of the copyright holder nor the names of
+	its contributors may be used to endorse or promote products
+	derived from this software without specific prior written permission.
+
+	THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+	CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+	INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+	OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+	DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+	CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+	SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+	NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+	LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+	HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+	OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+	SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import Foundation
+
+enum TestnetKeyServerError: Error {
+    case server(_ statusCode: Int, message: String?)
+    case invalidSignature
+    case allCoinsAllAllocated
+    case missingResponse
+    case responseInvalid
+    case unknown
+}
+
+struct TestnetServerRequest: Codable {
+    let signature: String
+    let public_nonce: String
+}
+
+class TestnetKeyServer {
+    private let MESSAGE_PREFIX = "Hello Tari from"
+    private let SERVER = "https://dropper.free.beeceptor.com" //TODO store in config
+    private let signature: Signature
+    private let url: URL
+    private let wallet: Wallet
+
+    init(wallet: Wallet) throws {
+        let (publicKey, publicKeyError) = wallet.publicKey
+        guard publicKeyError == nil else {
+            throw publicKeyError!
+        }
+
+        let (publicKeyHex, hexError) = publicKey!.hex
```

Comment:
here i think you used ! becaus you checkwd with guard if there is an error. So i guess it's fine.

Reply:
Yes it's fine, with TariLib if the error is nil, the result will never be nil